src/cleanfigure.m

```diff
@@ -672,113 +608,106 @@ function movePointsCloser(meta, handle)
-  dataNew = [];
-  lastReplIndex = 0;
-  lastEntryIsReplacement = false;
-  for k = 1:m
-     % Make sure that two subsequent moved points are separated by a NaN entry.
-     % This is to make sure that there is no visible line between two moved
-     % points that wasn't there before.
-     d = data(lastReplIndex+1:replaceIndices(k)-1,:);
-     if size(r{k}, 1) == 2
-         % Two replacement entries -- pad them with a NaN.
-         rep = [r{k}(1, :); ...
-                NaN(1, size(r{k}, 2)); ...
-                r{k}(2, :)];
-     else
-         rep = r{k};
-     end
-
-     % Don't draw line, if connecting line would be completely outside axis.
-     % We can check this using a line clipping algorithm.
-     % Illustration of the problem:
-     % http://www.cc.gatech.edu/grads/h/Hao-wei.Hsieh/Haowei.Hsieh/sec1_example.html
-     % This boils down to a line intersects line test, where all four lines of
-     % the axis rectangle need to be considered.
-     %
-     % First consider two easy cases:
-     % 1. This can't be the case, if last point was not replaced, because it is
-     %    inside the axis limits ('lastEntryIsReplacement == 0').
-     % 2. This can't be the case, if the current point will not be replace,
-     %    because it is inside the axis limits.
-     %    ( (isempty(d) && ~isempty(rep) == 0 ).
-     if lastEntryIsReplacement && (isempty(d) && ~isempty(rep))
-         % Now check if the connecting line goes through the axis rectangle.
-         % OR: Only do this, if the original segment was not visible either
-         bLineOutsideAxis = ~segmentVisible(...
-             data([lastReplIndex,replaceIndices(k)],:), ...
-             [false;false], xlim, ylim);
-
-         % If line is completly outside the axis, don't draw the line. This is
-         % achieved by adding a NaN and necessary, because the two points are
-         % moved close to the axis limits and thus would afterwards show a
-         % connecting line in the axis.
-         if bLineOutsideAxis
-             rep = [NaN(1, size(r{k}, 2)); rep];
-         end
-     end
-
-     % Add the data, depending if it is a valid point or a replacement
-     if ~isempty(d)     % Add current point from valid point 'd'
-         dataNew = [dataNew; d];
-         lastEntryIsReplacement = false;
-     end
-     if ~isempty(rep)   % Add current point from replacement point 'rep'
-         dataNew = [dataNew; rep];
-         lastEntryIsReplacement = true;
-     end
-
-     % Store last replacement index
-     lastReplIndex = replaceIndices(k);
+      % Add the new data for right points first, then the possible NaN and then
+      % the new left point
+      dataCell = cellfun(@(x, y) vertcat(x, y), dataCell, dataInsert_right, 'UniformOutput', false);
+      dataCell = cellfun(@(x, y) vertcat(x, y), dataCell, dataInsert_NaN,   'UniformOutput', false);
+      dataCell = cellfun(@(x, y) vertcat(x, y), dataCell, dataInsert_left,  'UniformOutput', false);
```

Comment:
Oh, I thought the point about cellfun was to avoid using loops. Your solution is indeed really nice